crates/pep508-rs/src/origin.rs

```diff
@@ -17,6 +19,7 @@ impl RequirementOrigin {
         match self {
             RequirementOrigin::File(path) => path.as_path(),
             RequirementOrigin::Project(path, _) => path.as_path(),
+            _ => panic!("Unsupported RequirementOrigin variant"),
```

Comment:
I understand that multiple toml files(pyproject.toml, uv.toml) are combined to obtain an override dependency.

It seems difficult to keep track of which files have which overriding dependencies.

I'll give it a thought.

Reply:
Make sense! What i'd like to do is try to avoid explicit panics where possible. Sometimes they are unavoidable, sometimes so you restructure code so the codepath becomes unreachable, sometimes you add a dummy value, etc.

Reply:
I changed to return dummy paths instead of panic.

https://github.com/astral-sh/uv/pull/3839/files#diff-a8cc5969fdef854f20c6a58c35e8727e9066c25b4625338501beda19b447ac2bR23